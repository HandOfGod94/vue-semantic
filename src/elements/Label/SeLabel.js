import Vue from 'vue'
import Component from 'vue-class-component'

@Component({
  name: 'se-label',
  props: {
    as: { type: String, default: 'div' },
    image: Boolean,
    color: String,
    basic: Boolean,
    empty: Boolean,
    circular: Boolean,
    size: String,
    corner: String,
    tag: Boolean,
    ribbon: Boolean,
    pointing: String,
    horizontal: Boolean,
    floating: Boolean,
    attach: String
  }
})
class SeLabel extends Vue {
  render (h) {
    let Label = this.as
    return (
      <Label class={this.classes}>
        {this.$slots.default}
      </Label>
    )
  }

  get classes () {
    let classes = {
      floating: this.floating,

      ui: true,

      [this.size]: this.size,
      image: this.image,
      tag: this.tag,
      ribbon: this.ribbon,
      [this.color]: this.color,
      empty: this.empty,
      horizontal: this.horizontal,
      circular: this.circular,
      basic: this.basic,
      [`${this.corner} corner`]: this.corner,
      [`${this.pointing} pointing`]: this.pointing,
      [`${this.attach} attached`]: this.attach,

      label: true
    }

    return classes
  }
}

export default SeLabel
