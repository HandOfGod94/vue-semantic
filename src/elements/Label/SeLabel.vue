<template>
  <component :is="as" :class="classObject">
    <slot></slot>
    <div v-if="hasDetail" class="detail">
      <slot name="detail">{{detail}}</slot>
    </div>
  </component>
</template>

<script>
export default {
  name: 'se-label',
  props: {
    as: { type: String, default: 'div' },
    image: Boolean,
    color: String,
    basic: Boolean,
    empty: Boolean,
    circular: Boolean,
    size: String,
    position: String,
    tag: Boolean,
    ribbon: Boolean,
    pointing: Boolean,
    horizontal: Boolean,
    floating: Boolean,
    attach: Boolean,
    detail: String
  },
  computed: {
    classObject () {
      return {
        floating: this.floating,
        ui: true,
        [this.size]: this.size,
        image: this.image,
        tag: this.tag,
        ribbon: this.ribbon,
        [this.color]: this.color,
        empty: this.empty,
        horizontal: this.horizontal,
        circular: this.circular,
        basic: this.basic,
        [this.position]: this.position,
        pointing: this.pointing,
        attached: this.attach,
        label: true
      }
    },
    hasDetail () { return !!this.$slots['detail'] || this.detail }
  }
}
</script>

<style>

</style>
