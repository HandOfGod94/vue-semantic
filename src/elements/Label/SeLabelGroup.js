import SeLabelGroupComponent from './SeLabelGroup.functional'

export default {
  name: 'se-label-group',
  props: {
    size: String,
    color: String,
    tag: Boolean,
    circurlar: Boolean
  },

  computed: {
    classObject () {
      return {
        ui: true,
        [this.size]: this.size,
        [this.color]: this.color,
        tag: this.tag,
        circurlar: this.circurlar,
        labels: true
      }
    }
  },

  render (createElement) {
    return createElement(SeLabelGroupComponent, {
      attrs: {
        slots: this.$slots
      },
      class: this.classObject
    })
  }
}
