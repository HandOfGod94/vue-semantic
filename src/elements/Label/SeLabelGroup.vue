<template>
  <div :class="classObject">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'se-label-group',
  props: {
    size: String,
    color: String,
    tag: Boolean,
    circurlar: Boolean
  },
  computed: {
    classObject () {
      return {
        ui: true,
        [this.size]: this.size,
        [this.color]: this.color,
        tag: this.tag,
        circurlar: this.circurlar,
        labels: true
      }
    }
  }
}

</script>

<style>

</style>
