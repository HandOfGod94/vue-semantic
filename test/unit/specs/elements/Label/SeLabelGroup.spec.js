import SeLabel from '@/elements/Label/SeLabel'
import SeLabelGroup from '@/elements/Label/SeLabelGroup'
import { mount } from '@vue/test-utils'

const labelWrapper = {
  render (h) {
    return (
      <SeLabel>Hello Label</SeLabel>
    )
  }
}

describe('SeLabelGroup Snapshot test', () => {
  it('should match with exisiting snapshot', () => {
    const data = { propsData: { circular: true }, slots: { default: labelWrapper } }
    const wrapper = mount(SeLabelGroup, data)
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('SeLabelGroup classes test', () => {
  it('should have "tag" class', () => {
    const data = { propsData: { tag: true }, slots: { default: labelWrapper } }
    const wrapper = mount(SeLabelGroup, data)
    expect(wrapper.classes()).toContain('tag')
  })

  it('should have correct size in classes', () => {
    const data = { propsData: { size: 'huge' }, slots: { default: labelWrapper } }
    const wrapper = mount(SeLabelGroup, data)
    expect(wrapper.classes()).toContain('huge')
  })
})
