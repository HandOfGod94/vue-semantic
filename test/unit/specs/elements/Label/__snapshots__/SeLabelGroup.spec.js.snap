// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeLabelGroup Snapshot test should match with exisiting snapshot 1`] = `
<div class="ui labels">
    <div class="ui label">Hello Label
        <!---->
    </div>
</div>
`;
